src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -465,24 +470,31 @@ citus_shard_cost_by_disk_size(PG_FUNCTION_ARGS)
 
 	if (queryResult != RESPONSE_OKAY)
 	{
+		MemoryContextSwitchTo(oldContext);
 		ereport(ERROR, (errcode(ERRCODE_CONNECTION_FAILURE),
 						errmsg("cannot get the size because of a connection error")));
 	}
 
 	List *sizeList = ReadFirstColumnAsText(result);
 	if (list_length(sizeList) != 1)
 	{
+		MemoryContextSwitchTo(oldContext);
```

Comment:
this is redundant, the memory context always needs to be switched by error handling logic.